genome_kit/_build_mane.py

```diff
@@ -0,0 +1,124 @@
+import os
+import pickle
+
+import genome_kit as gk
+import csv
+import gzip
+
+from genome_kit import gk_data
+from genome_kit._gk_data_config import get_mane_filename, get_mane_version
+
+_PRIORITY = {
+    "MANE Select": 0,
+    "MANE Plus Clinical": 1,
+}
+
+_SUPPORTED_ANNOS = {
+    "ncbi_refseq.hg38.p14_RS_2024_08": "1.4",
+    "gencode.v41": "1.0",
+}
+
+
+def _unversioned_id(id: str) -> str:
+    return id.rsplit(".", 1)[0]
+
+
+def build_mane(mane_version: str, genome: gk.Genome):
+    """Generates the data file that GK needs for fast retrieval of MANE Select transcripts.
+
+    Parameters
+    ----------
+    mane_version : :str
+        MANE version, e.g "1.4". see https://ftp.ncbi.nlm.nih.gov/refseq/MANE/MANE_human/
+    genome : :py:class:~genome_kit:Genome
+        A genome object with the matching annotations for the specified MANE version
+
+    Returns
+    -------
+    tuple
+        [1] a list of indices for :py:class:~genome_kit.Transcript objects for fast access to all
+            the transcripts that have a matching MANE Select transcript
+        [2] a list where the indices are that of the gene objects and the values are the index of the
+        corresponding MANE Select transcript in the genome.transcripts list.
+    """
+    mane_transcript_by_gene_idx = [None] * len(genome.genes)
+
+    # files originally downloaded from https://ftp.ncbi.nlm.nih.gov/refseq/MANE/MANE_human/
+    mane_src = gk.gk_data.get_file(f"MANE.GRCh38.v{mane_version}.summary.txt.gz")
+    with gzip.open(filename=mane_src, mode="rt") as f:
+        reader = csv.reader(f, delimiter="\t")
+        header = next(reader)
+        mane_status_idx = header.index("MANE_status")
+        transcript_id_col_idx = (
+            header.index("RefSeq_nuc")
+            if "ncbi_refseq" in genome.config
+            else header.index("Ensembl_nuc")
+        )
+        transcript_indices = set()
+
+        for line in reader:
+            # use only MANE Select (ignore MANE Plus Clinical)
```

Comment:
why?

Reply:
Based on user's request. Should be trivial to add MANE Plus Clinical if we want, I'm ok with it.

Reply:
Can we make this more clear in the comment or an issue?

Reply:
Sure, or I could just add support for Plus Clinical.